framework/transport/ieee1905_transport/ieee1905_transport_messages.cpp

```diff
@@ -19,6 +19,132 @@ namespace messages {
+
+static std::unique_ptr<Message>
+create_transport_message(Type type, std::initializer_list<messages::Message::Frame> frame)
+{
+    switch (messages::Type(type)) {
+    case messages::Type::CmduRxMessage:
+        return std::unique_ptr<messages::CmduRxMessage>{new messages::CmduRxMessage(frame)};
+    case messages::Type::CmduTxMessage:
+        return std::unique_ptr<messages::CmduTxMessage>{new messages::CmduTxMessage(frame)};
+    case messages::Type::SubscribeMessage:
+        return std::unique_ptr<messages::SubscribeMessage>{new messages::SubscribeMessage(frame)};
+    case messages::Type::CmduTxConfirmationMessage:
+        return std::unique_ptr<messages::CmduTxConfirmationMessage>{
+            new messages::CmduTxConfirmationMessage(frame)};
+    case messages::Type::InterfaceConfigurationQueryMessage:
+        return std::unique_ptr<messages::InterfaceConfigurationQueryMessage>{
+            new messages::InterfaceConfigurationQueryMessage(frame)};
+    case messages::Type::InterfaceConfigurationRequestMessage:
+        return std::unique_ptr<messages::InterfaceConfigurationRequestMessage>{
+            new messages::InterfaceConfigurationRequestMessage(frame)};
+    case messages::Type::InterfaceConfigurationIndicationMessage:
+        return std::unique_ptr<messages::InterfaceConfigurationIndicationMessage>{
+            new messages::InterfaceConfigurationIndicationMessage(frame)};
+    default:
+        LOG(WARNING) << "Received unknown message type: " << int(type);
+        return std::unique_ptr<messages::Message>{new messages::Message(0, frame)};
+    }
+}
+
+std::unique_ptr<Message> read_transport_message(Socket &sd)
+{
+    // Peek into the header to check if the entire message received
+    messages::Message::Header header;
+    auto bytes_ready = sd.getBytesReady();
+    auto read_bytes =
+        sd.readBytes(reinterpret_cast<uint8_t *>(&header), sizeof(header), false, sizeof(header));
+
+    if (read_bytes != sizeof(header)) {
+        LOG(ERROR) << "Error peeking into the message header: " << read_bytes;
+        return nullptr;
+    }
+
+    auto discard_pending_bytes = [&]() {
+        if (!sd.getBytesReady()) {
+            return;
+        }
+
+        // Discard all the "ready" bytes in the socket
+        // Reading an invalid message magic means that somehow the data synchronization
+        // was lost. Since the data is not necessarily aligned to any known size, we have
+        // two options here:
+        // 1. Safe - Discard 1 byte at a time, until finding the magic word
+        // 2. Faster - Discard sizeof(Header) bytes and hope to find a valid header afterwads
+        // 2. Fastest - Discard all the bytes and assume the sender will re-send the message
+        // For now we'll use the "Faster" method.
+        auto discarded_bytes = sd.readBytes(reinterpret_cast<uint8_t *>(&header), sizeof(header),
+                                            false, sd.getBytesReady());
```

Comment:
```suggestion
                                            false);
```
(the default "0" means that readBytes will use getBytesReady).

Reply:
Right, done.